### DISCUSSION TITLE
Automation Account Managed Identity as Admin to run API queries

### DISCUSSION THREAD

**Discussion by CCHV-GitHub** [Azure/ipam#130] [Q&A]
I'm using azure automation to run some API queries against Azure IPAM. 

The request I am using is the one to add a VNET to a block, as shown below

https://$appname.azurewebsites.net/api/spaces/$space/blocks/$blocks/networks

But the query is running as a Managed Identity on an Azure automation account and I get the following return message

{"error":"API restricted to admins."} 

However I can't see a way to add a managed identity as an admin in Azure IPAM. Is there another way I can achieve my goal here?

**Comment by CCHV-GitHub** [comment]
After some thought my own temp solution on this was to remove all administrators and then restrict access to the app using the Enterprise Application. However this does mean that everyone with access is now an admin but for now that's ok. I have raised a feature request in issues to allow managed identities as admins in the IPAM interface.

**Reply by DCMattyG** [reply]
Hi @CCHV-GitHub, I've responded to you in #131. We'll keep the conversation going there for continuity. Thanks!
